src/vcpkg/xunitwriter.cpp

```diff
@@ -4,58 +4,69 @@
 #include <vcpkg/commands.build.h>
 #include <vcpkg/xunitwriter.h>
 
-using namespace vcpkg;
-
-struct vcpkg::XunitTest
+namespace vcpkg
 {
-    std::string name;
-    std::string method;
-    std::string owner;
-    BuildResult result;
-    ElapsedTime time;
-    std::chrono::system_clock::time_point start_time;
-    std::string abi_tag;
-    std::vector<std::string> features;
-};
+    struct XunitTest
+    {
+        std::string name;
+        std::string method;
+        std::string owner;
+        CiResult ci_result;
+    };
+}
 
 namespace
 {
-    static void xml_test(XmlSerializer& xml, const XunitTest& test)
+    using namespace vcpkg;
+    void xml_test(XmlSerializer& xml, const XunitTest& test)
     {
         StringLiteral result_string = "";
-        switch (test.result)
+        switch (test.ci_result.code)
         {
+            case BuildResult::Succeeded: result_string = "Pass"; break;
+            case BuildResult::BuildFailed:
             case BuildResult::PostBuildChecksFailed:
-            case BuildResult::FileConflicts:
-            case BuildResult::BuildFailed: result_string = "Fail"; break;
+            case BuildResult::FileConflicts: result_string = "Fail"; break;
+            case BuildResult::CascadedDueToMissingDependencies:
             case BuildResult::Excluded:
-            case BuildResult::CascadedDueToMissingDependencies: result_string = "Skip"; break;
-            case BuildResult::Succeeded: result_string = "Pass"; break;
+            case BuildResult::ExcludedByParent:
+            case BuildResult::ExcludedByDryRun:
+            case BuildResult::Unsupported:
+            case BuildResult::Cached: // should this be "Pass" instead?
+                result_string = "Skip";
+                break;
+            case BuildResult::CacheMissing:
+            case BuildResult::Downloaded:
```

Comment:
It seems unfortunate to have so many lines dedicated to an assertion buried deep in the formatting logic, since the code flaw is likely much higher up the stack. No change requested.